plugin/validator_test.go

```diff
@@ -175,6 +179,160 @@ func TestValidator_ProcessAutoFields(t *testing.T) {
 	})
 }
 
+func TestValidator_SchemaAsJSON(t *testing.T) {
+	schemaNames := []string{
+		"acl",
+		"http-log",
+		"key-auth",
+		"rate-limiting",
+		"udp-log",
+	}
+	v, err := NewValidator()
+	assert.Nil(t, err)
+	for _, schemaName := range schemaNames {
+		schema, err := ioutil.ReadFile("testdata/" + schemaName + ".lua")
+		assert.Nil(t, err)
+		assert.Nil(t, v.LoadSchema(string(schema)))
+	}
+
+	t.Run("returns a valid JSON schema for loaded plugin schema", func(t *testing.T) {
+		for _, schemaName := range schemaNames {
+			schema, err := v.SchemaAsJSON(schemaName)
+			assert.Nil(t, err)
+			var kongPluginConfig KongPluginSchema
+			assert.Nil(t, json.Unmarshal([]byte(schema), &kongPluginConfig))
+		}
+	})
+
+	t.Run("validate schema for known good plugin [udp-log]", func(t *testing.T) {
+		schema, err := v.SchemaAsJSON("udp-log")
+		assert.Nil(t, err)
+		var kongPluginConfig KongPluginSchema
+		assert.Nil(t, json.Unmarshal([]byte(schema), &kongPluginConfig))
+		assert.EqualValues(t, 2, len(kongPluginConfig.Fields))
+
+		// Parse using DOM style with the map[string]interface{}
+		// This is simpler than trying to stuff a dynamic array into a static
+		// structure
```

Comment:
This is fine, it is just more code and a bit harder to grok. A simple assert.JSONeq would suffice as well. No changes required. This is good as is.

Reply:
yeah the grok is def a negative, we can always changes it later if we find it to be cumbersome in the future.